httomo/task_runner.py

```diff
@@ -243,6 +241,42 @@ def run_tasks(
         # TODO: After the blocks loop is complete the full data
         # is in "data_full_section" and the hdf5 file can be saved
         # or resliced if needed
+
+        # Assume that, after every section has been completed (except for the
+        # last section), a reslice should occur
+        if i < len(platform_sections) - 1:
+            reslice_info.count += 1
+            if reslice_info.count > 1 and not reslice_info.has_warn_printed:
+                reslice_warn_str = (
+                    "WARNING: Reslicing is performed more than once in this "
+                    "pipeline, is there a need for this?"
+                )
+                log_once(reslice_warn_str, comm=comm, colour=Colour.RED)
+                reslice_info.has_warn_printed = True
+
+            next_section_in = platform_sections[i+1].methods[0].parameters["data_in"]
+            # TODO: should `slicing_dim_section` be used to define
+            # `current_slice_dim`? Or would that be too confusing to randomly
+            # see `current_slice_dim = slicing_dim_section + 1`?
+            current_slice_dim = _get_slicing_dim(section.pattern)
+            next_slice_dim = _get_slicing_dim(platform_sections[i+1].pattern)
+            if reslice_dir is None:
+                resliced_data, _ = reslice(
+                    dict_datasets_pipeline[next_section_in],
+                    current_slice_dim,
+                    next_slice_dim,
+                    comm,
+                )
+            else:
+                resliced_data, _ = reslice_filebased(
+                    dict_datasets_pipeline[next_section_in],
+                    current_slice_dim,
+                    next_slice_dim,
+                    comm,
+                    reslice_dir,
+                )
+            # Store the resliced data
+            dict_datasets_pipeline[next_section_in] = resliced_data
```

Comment:
Can probably move some of the logic to a separate function `_perform_reslice()` or something, to make the sections-loop a bit nicer, I'll see how such a function ends up looking